### PR TITLE
Get Internship By Location [Created all the pages]

### LINKED ISSUE #255: Frontend of get Internships by location
- [x] Created page for get internship by location eg. for Delhi, Mumbai, India.
- [x] Connected to backend
- [x] Dynamic URL for this page

@pankajkumarbij @asi309 @palashgdev @Vidit-jindal assign me this issue.

### PR DESCRIPTION
**Pull request by EkjotKaur**
Fixed: #255 

# What I have done

- Created page for all internships
- Connected it to the backend
- Dynamic Rendering of pages
- Corrected the an mistake in frontend
- Student profile builder page is only visible then a student is logged in
-  All these pages are working

# Screenshots

###  All these pages are working

PR-43-9

One opening these pages

PR-43-4
PR-43-3
PR-43-1
PR-43-8
PR-43-7
PR-43-6
PR-43-5

@pankajkumarbij @asi309 @palashgdev @Vidit-jindal Please view my PR.

### CODE REVIEW

File: src/components/home/home.js [hunk 1]
```diff
@@ -102,44 +102,44 @@ const Home = () => {
...
-                  <a className="itemLink" href="/#">
+                  <a className="itemLink" href="/location-internship/Bangalore">
                     {" "}
                     <ListGroup.Item action variant="success">
                       Bangalore
                     </ListGroup.Item>
                   </a>
-                  <a className="itemLink" href="/#">
+                  <a className="itemLink" href="/location-internship/Chennai">
                     {" "}
```
Comment (himanshujaidka) [inline src/components/home/home.js:120]: why there is an empty list? In Pune and Mumbai, and in Delhi also, there is no empty list. line no. 120

**Comment by EkjotKaur** [comment]
@himanshujaidka When u will create internships in your database with location Pune, Mumbai, or Delhi then you can see the internships on these pages.

File: src/components/home/home.js [hunk 2]
```diff
@@ -102,44 +102,44 @@ const Home = () => {
...
                   </a>
-                  <a className="itemLink" href="/#">
+                  <a className="itemLink" href="/location-internship/Delhi">
                     <ListGroup.Item action variant="success">
                       Delhi
                     </ListGroup.Item>
                   </a>
-                  <a className="itemLink" href="/#">
+                  <a className="itemLink" href="/location-internship/India">
                     {" "}
```
Comment (himanshujaidka) [inline src/components/home/home.js:143]: I am talking about {""}, that in Delhi, Mumbai and Pune you are not initializing an empty list but in some parts you are using {""}, what is the purpose of this? line no. 143

**Comment by EkjotKaur** [comment]
Oh, that actually came automatically on auto formatting

**Comment by himanshujaidka** [comment]
> Oh, that actually came automatically on auto formatting

well, besides that, great work :) and pls change gugaon to gurugram

**Comment by EkjotKaur** [comment]
@pankajkumarbij  please view my PR.

**Review by pankajkumarbij — APPROVED** [review]

**Pull request merged**